### PR TITLE
bin/brew: repair style

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
repair `bin/brew` style for `brew style` with `--enable=all`
https://github.com/Homebrew/brew/blob/fa3af8d24d0335641e58a4355fa9de6d42ae4982/Library/Homebrew/style.rb#L188

related https://github.com/Homebrew/brew/issues/11086

### CODE REVIEW

File: bin/brew [hunk 1]
```diff
@@ -103,19 +111,20 @@ then
              http_proxy https_proxy ftp_proxy no_proxy all_proxy HTTPS_PROXY FTP_PROXY ALL_PROXY \
              GITHUB_ACTIONS GITHUB_WORKSPACE GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED \
              GITHUB_REPOSITORY GITHUB_RUN_ID GITHUB_SHA GITHUB_HEAD_REF GITHUB_BASE_REF GITHUB_REF \
-             "${!HOMEBREW_@}"
+             HOMEBREW_BREW_FILE HOMEBREW_EDITOR HOMEBREW_GITHUB_API_TOKEN HOMEBREW_LIBRARY HOMEBREW_PATH \
+             HOMEBREW_PREFIX HOMEBREW_REPOSITORY
```
Comment (Bo98) [inline bin/brew:115]: This is going to break a lot - like anything in `env_config.rb`.

File: bin/brew [hunk 2]
```diff
@@ -72,28 +72,36 @@ do
...
-  export "$VAR_NEW"="${!VAR}"
+  export "${VAR_NEW}"="${!VAR}"
 done
 
 export HOMEBREW_BREW_FILE
 export HOMEBREW_PREFIX
 export HOMEBREW_REPOSITORY
 export HOMEBREW_LIBRARY
 
+export VISUAL
```
Comment (MikeMcQuaid) [inline bin/brew:83]: Let's not add any new `export`s in here. Ignore them with a shellcheck ignore if needed.

File: bin/brew (hunk 1 above)
Comment (MikeMcQuaid) [reply bin/brew:115]: Yup. I'd just add a shellcheck ignore for this with a comment.

File: bin/brew [hunk 3]
```diff
@@ -72,28 +72,31 @@ do
...
-  export "$VAR_NEW"="${!VAR}"
+  export "${VAR_NEW}"="${!VAR}"
 done
 
 export HOMEBREW_BREW_FILE
 export HOMEBREW_PREFIX
 export HOMEBREW_REPOSITORY
 export HOMEBREW_LIBRARY
 
+# shellcheck disable=SC2154
```
Comment (MikeMcQuaid) [inline bin/brew:83]:
```suggestion
# set from user environment
# shellcheck disable=SC2154
```

File: bin/brew [hunk 4]
```diff
@@ -72,28 +72,31 @@ do
...
 # Use VISUAL if HOMEBREW_EDITOR and EDITOR are unset.
-if [[ -z "$HOMEBREW_EDITOR" && -n "$VISUAL" ]]
+if [[ -z "${HOMEBREW_EDITOR}" && -n "${VISUAL}" ]]
 then
-  export HOMEBREW_EDITOR="$VISUAL"
+  export HOMEBREW_EDITOR="${VISUAL}"
 fi
 
+# shellcheck disable=SC2154
```
Comment (MikeMcQuaid) [inline bin/brew:90]:
```suggestion
# set from user environment
# shellcheck disable=SC2154
```

File: bin/brew [hunk 5]
```diff
@@ -72,28 +72,31 @@ do
...
 # Set CI variable for Azure Pipelines and Jenkins
 # (Set by default on GitHub Actions, Circle and Travis CI)
-if [[ -z "$CI" ]] && [[ -n "$TF_BUILD" || -n "$JENKINS_HOME" ]]
+if [[ -z "${CI}" ]] && [[ -n "${TF_BUILD}" || -n "${JENKINS_HOME}" ]]
 then
   export CI="1"
 fi
 
-if [[ -z "$HOMEBREW_NO_ENV_FILTERING" ]]
+# shellcheck disable=SC2154
```
Comment (MikeMcQuaid) [inline bin/brew:98]:
```suggestion
# set from user environment
# shellcheck disable=SC2154
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @hyuraku!

**Pull request merged**